src/renderer/components/BackgroundMaps/BackgroundMapInfo.js

```diff
@@ -0,0 +1,163 @@
+// @ts-check
+import { Button, Fade, makeStyles, Paper, Typography } from '@material-ui/core'
+import * as React from 'react'
+import { defineMessages, useIntl } from 'react-intl'
+import DeleteIcon from '@material-ui/icons/DeleteForeverOutlined'
+
+import Loading from '../Loading'
+import { useMapServerQuery } from '../../hooks/useMapServerQuery'
+import { MapboxPrevOnly } from './MapCard'
+
+const m = defineMessages({
+  // Title for Offline Areas
+  offlineAreas: 'Offline Areas',
+  // Button to create an offline area
+  createOfflineArea: 'Create Offline Area',
+  // Button to delete style
+  deleteStyle: 'Delete Style',
+  // Title for error message when deleting style
+  deleteErrorTitle: 'Error Deleting Style',
+  // Description for error message when deleting style,
+  deleteErrorDescription: 'There was an error deleting the style',
+  // Zoom Level Title
+  zoomLevel: 'Zoom Level: {zoom}',
+  // abbreviation for megabyte
+  mb: 'mb'
+})
+
+/**
+ * @typedef BackgroundMapInfoProps
+ * @prop {string} id
+ * @prop {string} idBeingViewed
+ * @prop {string} url
+ * @prop {number} size
+ * @prop {()=>void} unsetMapValue
+ */
+
+/** @param {BackgroundMapInfoProps} props */
+export const BackgroundMapInfo = ({
+  id,
+  idBeingViewed,
+  unsetMapValue,
+  url,
+  size
+}) => {
+  const shouldLoad = React.useMemo(() => id === idBeingViewed, [
+    id,
+    idBeingViewed
+  ])
```

Comment:
yes, its for the fade effect. If we conditionally render using the parent, we still need to pass it some boolean to indicated that the fade effect should be run. So since we are already passing in the boolean, it seems to just handle all the loading state inside the component

Reply:
i see. aside from whether the fade effect is necessary at all, you don't need an explicit variable in this case to trigger the fade. specifying `<Fade in ... />` will work the same in this case.

Reply:
the `BackgroundMaps` component could be updated as such as a result:

```tsx
export const BackgroundMaps = ({ openSettings }) => {
  const { formatMessage: t } = useIntl()

  /** @type {MapServerStyleInfo['id']|false} */
  const initialMapId = /** {const} */ (false)

  const [mapValue, setMapValue] = React.useState(initialMapId)

  const { data } = useMapServerQuery('/styles')

  function unsetMapValue () {
    setMapValue(false)
  }

  const offlineMap = data && data.find(m => m.id === mapValue)

  return (
    <React.Fragment>
      <SidePanel
        mapValue={mapValue}
        openSettings={openSettings}
        setMapValue={setMapValue}
      />

      {!mapValue || !data ? (
        <div style={{ padding: 40 }}>
          <Typography variant='h4'> {t(m.mapBackgroundTitle)}</Typography>

          <Typography variant='body1'>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in
            reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
            pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
            culpa qui officia deserunt mollit anim id est laborum.
            <br />
          </Typography>
        </div>
      ) : offlineMap ? (
        <BackgroundMapInfo
          key={offlineMap.id}
          size={offlineMap.bytesStored}
          id={offlineMap.id}
          unsetMapValue={unsetMapValue}
          url={offlineMap.url}
        />
      ) : null}
    </React.Fragment>
  )
}
```

Reply:
https://github.com/digidem/mapeo-desktop/pull/678/commits/35536ba7d93b16a61bd742dc226353e375e9499c